File: frontend/src/types/analytics.ts
export type EmissionsByProject = {
  project_id: string;
  project_name: string;
  emissions: number;
};

export type EmissionsByMaterial = {
  material_name: string;
  emissions: number;
};

export type EmissionsByTime = {
  period_start: string;
  emissions: number;
};

File: app/schemas/analytics.py
from datetime import datetime

from pydantic import BaseModel


class EmissionsByProjectOut(BaseModel):
    project_id: str
    project_name: str
    emissions: float


class EmissionsByMaterialOut(BaseModel):
    material_name: str
    emissions: float


class EmissionsByTimeOut(BaseModel):
    period_start: datetime
    emissions: float
